packages/protoc-gen-es/src/protoc-gen-es-plugin.ts

```diff
@@ -263,8 +263,10 @@ function generateDescDoc(
 }
 
 // prettier-ignore
-function getFileDescCall(f: GeneratedFile, file: DescFile) {
-  const info = embedFileDesc(file);
+function getFileDescCall(f: GeneratedFile, file: DescFile, schema: Schema) {
+  const sourceFile = file.proto;
+  const runtimeFile = schema.proto.protoFile.find(f => f.name == sourceFile.name);
+  const info = embedFileDesc(runtimeFile ?? sourceFile);
```

Comment:
This removes source retention options from embedded file descriptors generated by protoc-gen-es. We simply do the reverse of packages/protoplugin/src/ecmascript/schema.ts:223.

Reply:
Might be worth a comment in here about why we want the version in `schema.proto.protoFile` instead of the provided `file.proto`, because it would already have omitted source-only options.

Reply:
Good call, added in https://github.com/bufbuild/protobuf-es/pull/827/commits/2ca5ebdfe34f78cab813ee7daa2be0a79bbd4a85